ENGAGEHF/Dashboard/Notifications/NotificationManager.swift

```diff
@@ -0,0 +1,158 @@
+    func registerSnapshotListener(user: User?) {
+        logger.info("Initializing notifiation snapshot listener...")
+        
+        // Remove previous snapshot listener for the user before creating new one
+        snapshotListener?.remove()
+        guard let uid = user?.uid else {
+            return
+        }
+        
+        let db = Firestore.firestore()
+        
+        // Ignore notifications older than expirationDate
+        guard let thresholdDate = Calendar.current.date(byAdding: .day, value: -expirationDate, to: .now) else {
+            logger.error("Unable to get threshold date: \(FetchingError.invalidTimestamp)")
+            return
+        }
+        
+        let thesholdTimeStamp = Timestamp(date: thresholdDate)
+        
+        // Set a snapshot listener on the query for valid notifications
+        db.collection("users").document(uid).collection("notifications")
+            .whereField("created", isGreaterThan: thesholdTimeStamp)
+            .addSnapshotListener { querySnapshot, error in
+                guard let documents = querySnapshot?.documents else {
+                    self.logger.error("Error fetching documents: \(error!)")
+                    return
+                }
+                
+                self.notifications = documents.compactMap {
+                    if $0.get("completed") == nil {
+                        return Notification(
+                            type: String(describing: $0["type"] ?? "Unknown"),
+                            title: String(describing: $0["title"] ?? "Unknown"),
+                            description: String(describing: $0["description"] ?? "Unknown"),
+                            id: $0.documentID
+                        )
+                    }
+                    return nil
+                }
+                
+                self.logger.debug("Notifications updated")
+            }
+    }
+    
+    func add(_ notification: Notification) async {
+        await standard.add(notification: notification)
+    }
+    
+    func markComplete(at offsets: IndexSet) async {
+        if ProcessInfo.processInfo.isPreviewSimulator {
+            notifications.remove(atOffsets: offsets)
+            return
+        }
+        
+        logger.debug("Marking notifications complete at the following offsets: \(offsets)")
+        
+        let db = Firestore.firestore()
+        
+        guard let user = Auth.auth().currentUser else {
+            logger.error("Unable to mark notitificaitons complete: \(FetchingError.userNotAuthenticated)")
+            return
+        }
+        
+        // Mark the notifications as completed in the Firestore
+        let timestamp = Timestamp(date: .now)
+        for offset in offsets {
+            let docID = notifications[offset].id
+            
+            let docRef = db.collection("users").document(user.uid).collection("notifications")
+                .document(docID)
```

Comment:
I would suggest a coherent formatting for elements like this. Either we have everything in a single line or spread across multiple lines.
```suggestion
            let docRef = db.collection("users")
                .document(user.uid)
                .collection("notifications")
                .document(docID)
```